### ISSUE TITLE
[REC] Fury Warrior - Anger Management - Rampage Incorrect Priority

### ISSUE THREAD

**Issue by Mosborne90** [Hekili/hekili#4201]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Spec

Warrior - Fury

### Describe the Issue

TLDR: Rampage is not being recommended properly.

I've been testing anger management using hekili and noticed some large issues which result in loss of DPS and inefficient cooldown reduction for our major abilities. Generally speaking, when running anger management, rampage should take priority over every minor ability with the exception of a 3 stack MFE execute. 

In multiple cases where fury should be using rampage to reduce cooldowns while talented into anger management, execute is recommended with a single MFE debuff, Raging Blow is recommend while capped on rage, and bloodthirst will sometimes appear over rampage. This results in a loss of cooldown reduction for anger management. 

**Execute recommended over rampage -** https://pastebin.com/tHwfyaj8

**Execute recommended over everything with no MFE stacks -** https://pastebin.com/6DGv8mjh

**Raging Blow recommended over rampage -** https://pastebin.com/6gERzAm4

**Bloodthirst recommended over rampage -** https://pastebin.com/UVgDEh0K

### How to Reproduce

Talent into AM, observe recommendations.

### Snapshot (Link)

https://pastebin.com/tHwfyaj8

### Raidbots Sim Report (Link)

https://www.raidbots.com/simbot/report/jQvEfCW9WFVyasuTA1RADi/simc

**Comment by Hekili** [comment]
### tl;dr: This matches the SimC priority, but the SimC priority should probably be updated.

> Execute recommended over rampage - [pastebin.com/tHwfyaj8](https://pastebin.com/tHwfyaj8)

Execute is above Rampage when any of the following are met:

```
        List ( slayer_am_st ) called from ( Fury:default:14 ) would PASS at 0.00.
        talent.slayers_dominance.enabled[true] & ( talent.anger_management.enabled[true] | ! talent.reckless_abandon.enabled[false] ) & active_enemies[1.00] = 1
         - this entry's criteria PASSES: debuff.marked_for_execution.stack[1.00] = 3 | buff.ashen_juggernaut.remains[0.00] < 2 | buff.sudden_death.stack[0.00] = 2 & buff.sudden_death.remains[0.00] < 7 | buff.sudden_death.remains[0.00] < 2
```

This passes because Ashen Juggernaut is down / Sudden Death is down.  This appears to match your sim, generally speaking, as this Execute is above all the Ramge entries in the priority.  You can see it is used 34.14 times in your sim before reaching Rampage entries.

\# | Count | action,conditions
-- | -- | --
V | 6.01 | recklessness,if=(!talent.anger_management&cooldown.avatar.remains<1&talent.titans_torment)\|talent.anger_management\|!talent.titans_torment
W | 3.75 | avatar,if=(talent.titans_torment&(buff.enrage.up\|talent.titanic_rage)&(debuff.champions_might.up\|!talent.champions_might))\|!talent.titans_torment
X | 7.09 | thunderous_roar,if=buff.enrage.up
Y | 7.03 | odyns_fury,if=dot.odyns_fury_torment_mh.remains<1&(buff.enrage.up\|talent.titanic_rage)&cooldown.avatar.remains
Z | **34.14** | **execute,if=debuff.marked_for_execution.stack=3\|buff.ashen_juggernaut.remains<2\|buff.sudden_death.stack=2&buff.sudden_death.remains<7\|buff.sudden_death.remains<2**
a | 6.88 | rampage,if=talent.bladestorm&cooldown.bladestorm.remains<=gcd&!debuff.champions_might.up
b | 9.80 | bladestorm,if=buff.enrage.up&(cooldown.recklessness.remains>=9\|cooldown.avatar.remains>=9)
c | 7.13 | rampage,if=!buff.enrage.up
d | 47.89 | rampage,if=action.raging_blow.charges<=1&rage>=115
e | 26.97 | bloodthirst,if=buff.bloodcraze.stack>3\|crit_pct_current>=85
f | 131.88 | raging_blow
g | 18.63 | bloodthirst
  | 0.00 | execute
h | 11.99 | rampage
i | 7.45 | whirlwind,if=talent.meat_cleaver

### That being said...

I am guessing that your sim needs revisions:
* You don't have Ashen Juggernaut, so it's always going to have < 2 seconds remaining; and
* The check for Sudden Death *probably* intends that Sudden Death is actually up when it checks its remaining time (i.e., trying not to let Sudden Death go to waste).

Changing that line in the priority to...
```
actions.slayer_am_st+=/execute,if=debuff.marked_for_execution.stack=3|talent.ashen_juggernaut&buff.ashen_juggernaut.remains<2|buff.sudden_death.stack=2&buff.sudden_death.remains<7|buff.sudden_death.up&buff.sudden_death.remains<2
```
...gives us [raidbots.com/simbot/report/jvjn5y71Z9WHHPg8mvU1TP/simc](https://www.raidbots.com/simbot/report/jvjn5y71Z9WHHPg8mvU1TP/simc).

![image](https://github.com/user-attachments/assets/abbcb1e7-c18d-4788-b419-6974ad4e807d)

This is a 1.2% DPS improvement in your current specialization.

\# | count | action,conditions
-- | -- | --
Z | **18.52** | **execute,if=debuff.marked_for_execution.stack=3\|talent.ashen_juggernaut&buff.ashen_juggernaut.remains<2\|buff.sudden_death.stack=2&buff.sudden_death.remains<7\|buff.sudden_death.up&buff.sudden_death.remains<2**
a | 6.69 | rampage,if=talent.bladestorm&cooldown.bladestorm.remains<=gcd&!debuff.champions_might.up
c | 6.04 | rampage,if=!buff.enrage.up
d | 51.22 | rampage,if=action.raging_blow.charges<=1&rage>=115
e | 28.33 | bloodthirst,if=buff.bloodcraze.stack>3\|crit_pct_current>=85
f | 139.09 | raging_blow
g | 21.33 | bloodthirst
h | 5.12 | execute
i | 11.46 | rampage

This results in ~15 fewer Executes in this higher priority position, 10 fewer Executes overall.

### Comparison
Action | DPS | DPS% | Execute | Interval | DPE | DPET
-- | -- | -- | -- | -- | -- | --
Execute | 0 (108,724) | 0.0% (8.2%) | 34.2 | 8.71s | 956,747 | 1,053,239
Execute (Modified) | 0 (90,905) | 0.0% (6.8%) | 23.6 | 12.62s | 1,159,404 | 1,286,016
Rampage | 0 (224,312) | 0.0% (17.0%) | 73.8 | 4.08s | 911,807 | 1,002,762
Rampage (Modified) | 0 (230,768) | 0.0% (17.3%) | 75.4 | 4.02s | 922,649 | 1,017,261

**Comment by Mosborne90** [comment]
> ### tl;dr: This matches the SimC priority, but the SimC priority should probably be updated.

I could be asking too much from the addon honestly. I noticed this issue on mythic broodtwister progression where you need to have major cooldowns for each add set. Following hekili will bring back major cooldowns within 55s-1m while prioritizing rampage above all else brings this to 45-47 seconds. slower cooldown timing means adds die before your cooldowns are back and then you've lost a massive portion of damage. This type of event won't really come up on a sim since we can't sim a particular boss. I appreciate you taking a look at this but it may not be worth an update unless some kind of feature can be added to prioritize cooldown reduction when talented into anger management. Thank you again, keep up the great work!

**Comment by Hekili** [comment]
I don't think you understand that I gathered data and ultimately agreed with you.

**Comment by Mosborne90** [comment]
> I don't think you understand that I gathered data and ultimately agreed with you.

yeah apologies. I don't know who actually creates these APLs and I'm not experienced with editing my own. Felt like I was getting in over my head with this recommendation. That being said, is this a change you would make? "talent.ashen_juggernaut" seems like a check that should have been in place to start. Thank you again for taking the time.

**Comment by Mosborne90** [comment]
The changes made today have made some impact on cooldown reduction for major abilities. Recklessness is now coming up within 50 seconds, down from 55s-1m, and execute is being recommended properly. The only remaining issue is that raging blow is being recommended while capped on rage instead of rampage. 

This seems to only happen if you get an enrage buff which causes rampage to fall to a low priority, regardless of your current rage amount. The result of this is basically less rage spent overall, which explains the slightly longer cooldown recovery.

You can see this raging blow recommendation here: https://pastebin.com/xy9gYF1w

**Comment by syrifgit** [comment]
Closing as this ticket is 4 versions old now, can reopen with new snapshots if it persists.
